modules/ROOT/pages/dataweave-extension-plugin.adoc

```diff
@@ -0,0 +1,362 @@
+
+image::dataweave-extension-dataweavedependencies.png[DataWeave Dependencies View]
+
+
+== Deploy DataWeave Libraries
+
+Before deploying your DataWeave library to Anypoint Exchange, generate documentation for your mapping or module functions. When you deploy your DataWeave library to Exchange, the documentation is also published to the Exchange library page.
+
+To generate documentation for the functions, follow these steps:
+
+. In Visual Studio Code, open your DataWeave project.
+. Navigate to the `src/main/dw` folder.
+. Click the *Generate Weave Documentation* button located at the top of the function in the `.dwl` file.
++
+image::dataweave-extension-generatedocs-1.png[MyMapping.dwl file and the Generate Weave Documentation button]
+
+The documentation template appears as a comment above the function:
+
+image::dataweave-extension-generatedocs-2.png[Auto-generated documentation located above the function]
+
+To deploy a DataWeave library to Exchange:
+
+. In Visual Studio Code, open your DataWeave project.
+. Open and configure your `pom.xml` file.
+.. Ensure that your `groupId` is set to your `organization ID`.
+.. Add the Maven facade as a repository within the `<distributionManagement>` element.
++
+image::dataweave-extension-mavenfacade.png[Maven facade in <distributionManagement> element]
+
+[start=3]
+. Update the `settings.xml` file in your Maven `.m2` directory with your Anypoint Platform credentials.
++
+For guidance, refer to the xref:exchange::to-publish-assets-maven.adoc#publish-an-asset-to-exchange-using-maven guide[Publish an Asset to Exchange Using Maven] documentation.
+
+[start=4]
+. Execute the Maven `deploy` command to upload the DataWeave library to the deployment target and the auto-generated documentation to Exchange.
++
+For guidance, refer to the xref:dataweave-maven-plugin.adoc[DataWeave Maven Plugin] documentation.
+
+== Consume DataWeave Libraries
+
+After deploying your DataWeave library to Exchange, the library appears as an asset for your organization in the Exchange marketplace.
+
+To consume the DataWeave library:
+
+. Add the library’s group ID, artifact ID, version, and classifier to the `dependencies` element in your project’s `pom.xml` file.
++
+You can copy the dependency snippet from Exchange.
+
+. Add the Maven facade as a repository in the `repositories` section.
+
+For more details, refer to the xref:exchange::to-publish-assets-maven.adoc#publish-and-consume-federated-assets[Publish and Consume Federated Assets] documentation.
+
+
+== DataWeave 2.0 Extension Language Edition Features
+
+The Language Edition features of the DataWeave 2.0 extension include completion, navigation, code editing, and code inspection.
+
+=== Completion
+
+The completion feature enables you to autocomplete content for any visible functions, variables, or types. The feature also suggests fields based on the type inference.
+
+image::dataweave-extension-completion.png[Completion Feature sample]
```

Comment:
Can we remove the input directive?
And also remove the parameters? because that is autocompleted too when you call to `filter`